File: ruby/bob/bob.rb
class Bob
  def hey(phrase)
    phrase = phrase.gsub(/[^A-Za-z0-9?]/,'')
    if phrase == phrase.upcase && phrase.gsub(/[^A-Z]/,'').length > 0
      return 'Whoa, chill out!'
    end
    return 'Sure.' if phrase[-1] == '?'
    return 'Fine. Be that way!' if phrase.empty?
    'Whatever.'
  end
end

File: javascript/bob/bob.js

var Bob = function() {

  this.hey = function(input) {
    var up = input.toUpperCase();
    if (input.replace(/^\s+|\s+$/g, '') == '') { return 'Fine. Be that way!'}
    if (up == input && up != input.toLowerCase()) {
      return 'Whoa, chill out!'
    }
    if (input[input.length - 1] == '?') { return 'Sure.' }
    return 'Whatever.';
  }

};

module.exports = Bob;
